fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerIndexFromIndexTest.java

```diff
@@ -0,0 +1,365 @@
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Tests for building indexes from other indexes with {@link OnlineIndexer}.
+ */
+
+public class OnlineIndexerIndexFromIndexTest extends OnlineIndexerTest {
+
+    @Test
+    public void testIndexFromIndexSimple() {
+
+        final FDBStoreTimer timer = new FDBStoreTimer();
+        final long numRecords = 1000;
+
+        List<TestRecords1Proto.MySimpleRecord> records = LongStream.range(0, numRecords).mapToObj(val ->
+                TestRecords1Proto.MySimpleRecord.newBuilder().setRecNo(val).build()
+        ).collect(Collectors.toList());
+        Index srcIndex = new Index("src_index", field("num_value_2"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS);
+        Index tgtIndex = new Index("tgt_index", field("num_value_3_indexed"), IndexTypes.VALUE);
+
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = metaDataBuilder -> {
+            metaDataBuilder.addIndex("MySimpleRecord", srcIndex);
+            metaDataBuilder.addIndex("MySimpleRecord", tgtIndex);
+        } ;
+
+        openSimpleMetaData();
+        try (FDBRecordContext context = openContext())  {
+            records.forEach(recordStore::saveRecord);
+            context.commit();
+        }
+
+        openSimpleMetaData(hook);
+        try (FDBRecordContext context = openContext()) {
+            try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                    .setDatabase(fdb).setMetaData(metaData).setIndex(srcIndex).setSubspace(subspace)
+                    .build()) {
+                recordStore.markIndexWriteOnly(srcIndex).join();
+                indexBuilder.rebuildIndex(recordStore);
+                recordStore.markIndexReadable(srcIndex).join();
+                context.commit();
+                assertTrue(recordStore.isIndexReadable(srcIndex));
+            }
+        }
+
+        openSimpleMetaData(hook);
+        try (FDBRecordContext context = openContext()) {
+            recordStore.markIndexWriteOnly(tgtIndex).join();
+            context.commit();
+            assertTrue(recordStore.isIndexWriteOnly(tgtIndex));
+        }
+
+        openSimpleMetaData(hook);
+        try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setIndex(tgtIndex).setSubspace(subspace)
+                .setIndexFromIndex(new OnlineIndexer.IndexFromIndexRequest("src_index")
+                        .mustUseSourceIndex())
+                .setTimer(timer)
+                .build()) {
+
+            indexBuilder.buildIndexAsync(true)
+                    .handle((ignore, e) ->
+                    {
+                        assertNull(e);
+                        return null;
+                    })
+                    .join();
```

Comment:
```suggestion
            indexBuilder.buildIndex();
```

I think the assert here is basically just validating that there aren't errors, and if there are, those will bubble up through `buildIndex` and fail the test.